src/sentry_core.c

```diff
@@ -114,7 +114,15 @@ sentry_shutdown(void)
     if (options) {
         if (options->transport && options->transport->shutdown_func) {
             SENTRY_TRACE("shutting down transport");
-            options->transport->shutdown_func(options->transport);
+            // TODO: make this configurable
```

Comment:
How about we add this to the transport struct and expose a setter, just like the `send_envelope_func`, etc? Looks like a small enough change to do it in this PR. Note that there should also be a way to disable the timeout completely.

Reply:
According to unified API, this lives in the generic init options. In either case I would rather do that as a followup.

Reply:
Ah, interesting. I had only compared with Rust where this was on the transport.